runtime-modules/bounty/src/benchmarking.rs

```diff
@@ -80,51 +93,150 @@ fn member_funded_account<T: Trait + membership::Trait>(
+        account_id.clone(),
+    )
+    .unwrap();
+
+    (account_id, new_member_id)
+}
+
+fn announce_entry_and_submit_work<T: Trait + membership::Trait>(
+    bounty_id: &T::BountyId,
+    index: u32,
+) -> T::EntryId {
+    let membership_index = 1000 + index;
+    let (account_id, member_id) = member_funded_account::<T>("work entrants", membership_index);
+
+    Bounty::<T>::announce_work_entry(
+        RawOrigin::Signed(account_id.clone()).into(),
+        member_id,
+        *bounty_id,
+        account_id.clone(),
+    )
+    .unwrap();
+
+    let entry_id: T::EntryId = Bounty::<T>::entry_count().into();
+
+    let work_data = b"work_data".to_vec();
+
+    Bounty::<T>::submit_work(
+        RawOrigin::Signed(account_id.clone()).into(),
+        member_id,
+        *bounty_id,
+        entry_id,
+        work_data,
+    )
+    .unwrap();
+
+    entry_id
+}
+
+fn create_max_funded_bounty<T: Trait>(params: BountyCreationParameters<T>) -> T::BountyId {
+    let funding_amount = match params.funding_type {
+        FundingType::Perpetual { target } => target,
+        FundingType::Limited {
+            max_funding_amount, ..
+        } => max_funding_amount,
+    };
+
+    create_funded_bounty::<T>(params.clone(), funding_amount)
+}
+
+fn create_min_funded_bounty<T: Trait>(params: BountyCreationParameters<T>) -> T::BountyId {
+    let funding_amount = match params.funding_type {
+        FundingType::Perpetual { target } => target,
+        FundingType::Limited {
+            min_funding_amount, ..
+        } => min_funding_amount,
+    };
+
+    create_funded_bounty::<T>(params.clone(), funding_amount)
+}
+
+fn create_funded_bounty<T: Trait>(
+    params: BountyCreationParameters<T>,
+    funding_amount: BalanceOf<T>,
+) -> T::BountyId {
+    T::CouncilBudgetManager::set_budget(params.cherry + funding_amount);
+
+    Bounty::<T>::create_bounty(RawOrigin::Root.into(), params, Vec::new()).unwrap();
 
-    (account_id, T::MemberId::from(id.saturated_into()))
+    let bounty_id: T::BountyId = Bounty::<T>::bounty_count().into();
+
+    assert!(Bounties::<T>::contains_key(bounty_id));
+
+    Bounty::<T>::fund_bounty(
+        RawOrigin::Root.into(),
+        BountyActor::Council,
+        bounty_id,
+        funding_amount,
+    )
+    .unwrap();
+
+    bounty_id
 }
 
 const MAX_BYTES: u32 = 50000;
 const SEED: u32 = 0;
+const MAX_WORK_ENTRIES: u32 = 100;
 
 benchmarks! {
     where_clause {
         where T: council::Trait,
               T: balances::Trait,
               T: membership::Trait,
+              T: Trait,
     }
     _{ }
 
     create_bounty_by_council {
         let i in 1 .. MAX_BYTES;
         let metadata = vec![0u8].repeat(i as usize);
         let cherry: BalanceOf<T> = 100.into();
+        let entrant_stake: BalanceOf<T> = T::MinWorkEntrantStake::get();
+        let max_amount: BalanceOf<T> = 1000.into();
 
         T::CouncilBudgetManager::set_budget(cherry);
 
         let params = BountyCreationParameters::<T>{
```

Comment:
I think this should also have a `Closed` contract type since it's the worst case scenario for `ensure_create_bounty_parameters_valid`.

Reply:
OK.

Reply:
I think it'd be a good idea to have a `let j in 1 .. T::ClosedContractSizeLimit::get()` for the number of members, we can use that to calculate the weight in the extrinsic.